src/Aquifer.Common/Services/Caching/CachingLanguageService.cs

```diff
@@ -57,11 +70,113 @@ public async Task<IReadOnlyDictionary<string, int>> GetLanguageIdByCodeMapAsync(
                 cacheEntry.SlidingExpiration = s_cacheLifetime;
 
                 var LanguageCodeById = await _dbContext.Languages
-                    .ToDictionaryAsync(li => li.Id, li => li.ISO6393Code, ct);
+                    .Select(l => new { l.Id, l.ISO6393Code })
```

Comment:
Interesting performance discovery: EF isn't smart enough to use the `ToDictionary()` properties to inform the columns in the SELECT.  You can force it to only retrieve the ones you care about with a manual `.Select()`.